packages/components/src/components/checkboxgroup/checkboxgroup.component.ts

```diff
@@ -0,0 +1,84 @@
+import { CSSResult, html, nothing } from 'lit';
+import { property } from 'lit/decorators.js';
+import { Component } from '../../models';
+import { TAG_NAME as CHECKBOX_NAME } from '../checkbox/checkbox.constants';
+import { TYPE as TEXT_TYPE, VALID_TEXT_TAGS as TEXT_TAGS } from '../text/text.constants';
+import styles from './checkboxgroup.styles';
+
+/**
+ * `mdc-checkboxgroup` component allows you to select multiple options from a predefined list.
+ * It is commonly used in forms where multiple selections are required, such as preferences, filters, or categories.
+ *
+ * A checkbox group typically consists of multiple checkboxes grouped together,
+ * each representing a selectable option. You can check or uncheck options based on their preferences.
+ *
+ * @dependency mdc-text
+ *
+ * @tagname mdc-checkboxgroup
+ *
+ * @slot default - This is a default slot
+ */
+class Checkboxgroup extends Component {
+  /**
+   * The header text of the checkboxgroup.
+   */
+  @property({ type: String, attribute: 'header-text' }) headerText?: string;
+
+  /** @internal */
+  @property({ type: String, reflect: true }) override role = 'group';
+
+  /** @internal */
+  private get checkboxList() {
+    const slot = this.shadowRoot?.querySelector('slot');
+    const assignedElements = slot?.assignedElements() || [];
+    return assignedElements.filter((element) => element.tagName.toLowerCase() === CHECKBOX_NAME);
```

Comment:
```suggestion
    return assignedElements.filter((element) => element.tagName.toLowerCase() === CHECKBOX_TAGNAME);
```

`CHECKBOX_TAGNAME` can be stored in constants file and point to `TAG_NAME` from checkbox.constants.ts

In this way, if we make any changes to the tagname further, it would automatically reflect here.